catalog-rest-service/src/main/java/org/openmetadata/catalog/resources/services/pipeline/PipelineServiceResource.java

```diff
@@ -0,0 +1,202 @@
+
+  public static EntityReference addHref(UriInfo uriInfo, EntityReference service) {
+    return service.withHref(RestUtil.getHref(uriInfo, "v1/services/pipelineServices/", service.getId()));
+  }
+
+  private static List<PipelineService> addHref(UriInfo uriInfo, List<PipelineService> instances) {
+    instances.forEach(i -> addHref(uriInfo, i));
+    return instances;
+  }
+
+  private static PipelineService addHref(UriInfo uriInfo, PipelineService pipelineService) {
+    pipelineService.setHref(RestUtil.getHref(uriInfo, "v1/services/pipelineServices/",
+            pipelineService.getId()));
+    return pipelineService;
+  }
+
+  @Inject
+  public PipelineServiceResource(PipelineServiceRepository dao, CatalogAuthorizer authorizer) {
+    Objects.requireNonNull(dao, "PipelineServiceRepository must not be null");
+    this.dao = dao;
+    this.authorizer = authorizer;
+  }
+
+  static class PipelineServiceList extends ResultList<PipelineService> {
+    PipelineServiceList(List<PipelineService> data) {
+      super(data);
+    }
+  }
+
+  @GET
+  @Operation(summary = "List Pipeline services", tags = "services",
+          description = "Get a list of pipeline services.",
+          responses = {
+                  @ApiResponse(responseCode = "200", description = "List of pipeline service instances",
+                          content = @Content(mediaType = "application/json",
+                          schema = @Schema(implementation = PipelineServiceList.class)))
+          })
+  public PipelineServiceList list(@Context UriInfo uriInfo, @QueryParam("name") String name) throws IOException {
+    return new PipelineServiceList(addHref(uriInfo, dao.list(name)));
+  }
+
+  @GET
+  @Path("/{id}")
+  @Operation(summary = "Get a pipeline service", tags = "services",
+          description = "Get a pipeline service by `id`.",
+          responses = {
+                  @ApiResponse(responseCode = "200", description = "Pipeline service instance",
+                          content = @Content(mediaType = "application/json",
+                          schema = @Schema(implementation = PipelineService.class))),
+                  @ApiResponse(responseCode = "404", description = "Pipeline service for instance {id} is not found")
+          })
+  public  PipelineService get(@Context UriInfo uriInfo,
+                             @Context SecurityContext securityContext,
+                             @PathParam("id") String id) throws IOException {
+    return addHref(uriInfo, dao.get(id));
+  }
+
+  @GET
+  @Path("/name/{name}")
+  @Operation(summary = "Get messaging service by name", tags = "services",
+          description = "Get a messaging service by the service `name`.",
+          responses = {
+                  @ApiResponse(responseCode = "200", description = "Messaging service instance",
+                          content = @Content(mediaType = "application/json",
+                          schema = @Schema(implementation = Dashboard.class))),
```

Comment:
Pipeline.class instead of dashboard.class

Reply:
Fixed it